otx/algorithms/segmentation/adapters/mmseg/models/losses/detcon_loss.py

```diff
@@ -0,0 +1,157 @@
+            dist.all_gather(target1_large, target1)
+            dist.all_gather(target2_large, target2)
+            target1_large = torch.cat(target1_large, dim=0)
+            target2_large = torch.cat(target2_large, dim=0)
+
+            # Fold into batch dimension
+            target1_large = target1_large.reshape(-1, num_samples, num_features)
+            target2_large = target2_large.reshape(-1, num_samples, num_features)
+
+            # Create the labels by using the current replica ID and offsetting.
+            replica_id = dist.get_rank()
+            labels_idx = torch.arange(batch_size) + replica_id * batch_size
+            enlarged_bs = target1_large.shape[0]
+            labels = F.one_hot(labels_idx, num_classes=enlarged_bs).to(pred1.device)
+        else:
+            target1_large = target1
+            target2_large = target2
+            labels = F.one_hot(torch.arange(batch_size), num_classes=batch_size).to(pred1.device)
+
+        labels = labels.unsqueeze(dim=2).unsqueeze(dim=1)
+
+        # Do our matmuls and mask out appropriately.
+        logits_aa = torch.einsum("abk,uvk->abuv", pred1, target1_large) / self.temperature
+        logits_bb = torch.einsum("abk,uvk->abuv", pred2, target2_large) / self.temperature
+        logits_ab = torch.einsum("abk,uvk->abuv", pred1, target2_large) / self.temperature
+        logits_ba = torch.einsum("abk,uvk->abuv", pred2, target1_large) / self.temperature
+
+        labels_aa = labels * same_obj_aa
+        labels_ab = labels * same_obj_ab
+        labels_ba = labels * same_obj_ba
+        labels_bb = labels * same_obj_bb
+
+        logits_aa = logits_aa - infinity_proxy * labels * same_obj_aa
+        logits_bb = logits_bb - infinity_proxy * labels * same_obj_bb
+        labels_aa = 0.0 * labels_aa
+        labels_bb = 0.0 * labels_bb
+        if not local_negatives:
+            logits_aa = logits_aa - infinity_proxy * labels * (1 - same_obj_aa)
+            logits_ab = logits_ab - infinity_proxy * labels * (1 - same_obj_ab)
+            logits_ba = logits_ba - infinity_proxy * labels * (1 - same_obj_ba)
+            logits_bb = logits_bb - infinity_proxy * labels * (1 - same_obj_bb)
+
+        labels_abaa = torch.cat([labels_ab, labels_aa], dim=2)
+        labels_babb = torch.cat([labels_ba, labels_bb], dim=2)
+
+        labels_0 = labels_abaa.reshape((batch_size, num_samples, -1))
+        labels_1 = labels_babb.reshape((batch_size, num_samples, -1))
+
+        num_positives_0 = torch.sum(labels_0, dim=-1, keepdim=True)
+        num_positives_1 = torch.sum(labels_1, dim=-1, keepdim=True)
+
+        labels_0 = labels_0 / torch.maximum(num_positives_0, torch.tensor(1.0, device=num_positives_0.device))
+        labels_1 = labels_1 / torch.maximum(num_positives_1, torch.tensor(1.0, device=num_positives_0.device))
+
+        obj_area_0 = torch.sum(make_same_obj(pind1, pind1), dim=(2, 3))
+        obj_area_1 = torch.sum(make_same_obj(pind2, pind2), dim=(2, 3))
```

Comment:
can't we reuse `same_obj_aa` and `same_obj_bb` for here?

Reply:
`same_obj_aa` is for `pind1` and `tind1`, and `same_obj_bb` is for `pind2` and `tind2`. ([link](https://github.com/openvinotoolkit/training_extensions/blob/feature/otx-warmstart-seg/otx/algorithms/segmentation/adapters/mmseg/models/losses/detcon_loss.py#L69-L72))
So, `make_same_obj(pind1, pind1)` must be newly set to get `obj_area_0`, and so `make_same_obj(pind2, pind2)`.